crypto/bls12381/bls.go

```diff
@@ -59,6 +59,28 @@ func Verify(sig Signature, pk PublicKey, msg []byte) (bool, error) {
 	return dummySig.VerifyCompressed(sig, false, pk, false, msg, DST), nil
 }
 
+// PopProve signs on a msg using a BLS secret key for proof-of-possession
+// using DST_POP
+func PopProve(sk PrivateKey, msg []byte) Signature {
+	secretKey := new(blst.SecretKey)
+	secretKey.Deserialize(sk)
+	return new(BlsSig).Sign(secretKey, msg, DST_POP).Compress()
+}
+
+// PopVerify verifies a BLS sig generated from PopProve over msg with a
+// BLS public key. The sig and public key are all compressed
+func PopVerify(sig Signature, pk PublicKey, msg []byte) (bool, error) {
+	dummySig := new(BlsSig)
+	return dummySig.VerifyCompressed(sig, false, pk, false, msg, DST_POP), nil
```

Comment:
shouldn't we verify the:
- sigGroupcheck
- pkValidate

i.e set both check flags to `true` ?

Reply:
Nice find! Although not sure `pkValidate` should be true as it is not mentioned by Cubist. Will check with them